src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,28 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Transaction transaction = null;
+        try (Session session = factory.openSession()) {
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can`t create movie " + movie, e);
+        }
     }
 
     @Override
     public Optional<Movie> get(Long id) {
-        return null;
+        Movie movie = null;
+        try (Session session = factory.openSession()) {
+            movie = session.get(Movie.class, id);
+        } catch (Exception e) {
+            throw new DataProcessingException("Can`t get movie by Id " + id, e);
+        }
+        return Optional.ofNullable(movie);
```

Comment:
I think it would be better to put return in the try block and also initialize the movie there